CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/importing/file/GraphMLImportProcessor.java

```diff
@@ -178,20 +187,42 @@ public void process(final PluginParameters parameters, final File input, final R
                                     }
                                     //store the type of each node id so that an edge can be matched to the correct source and destination
                                     if (retrieveTransactions) {
-                                        nodeIdToType.put(id, nodeRecords.get(GraphRecordStoreUtilities.SOURCE + AnalyticConcept.VertexAttribute.TYPE));
+                                        nodeIdToType.put(stringID, nodeRecords.get(GraphRecordStoreUtilities.SOURCE + AnalyticConcept.VertexAttribute.TYPE));
                                     }
                                     break;
                                 }
                                 case EDGE_TAG: {
                                     if (retrieveTransactions) {
                                         final NamedNodeMap attributes = childNode.getAttributes();
-                                        final String id = attributes.getNamedItem(ID_TAG).getNodeValue();
-                                        final String source = attributes.getNamedItem(EDGE_SRC_TAG).getNodeValue();
-                                        final String target = attributes.getNamedItem(EDGE_DST_TAG).getNodeValue();
+                                        Node id = attributes.getNamedItem(ID_TAG);
+
+                                        /* Constellation requires edge Idenitifers 
+                                        ** but this isn't a requirement in the graphML specification
+                                        ** http://graphml.graphdrawing.org/specification/xsd.html
+                                        ** If no edge ID is given a UUID will be geenrated and assigned.
+                                        */
+                                        final String stringID = (id == null) ? UUID.randomUUID().toString() : id.getNodeValue();
```

Comment:
is there any risk that `randomUUID()` inadvertently randomly generates the same value for 2 different transactions? Just wondering if there's a chance (however small) that transactions get unintentionally merged together

Reply:
There is but it's very small.

>  The number of random version-4 UUIDs which need to be generated in order to have a 50% probability of at least one collision is 2.71 quintillion.
[Wikipedia - Universally_unique_identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier#Random_UUID_probability_of_duplicates)

Reply:
Yeah ok those odds seems low enough 😆